File: 0238-product-of-array-except-self/0238-product-of-array-except-self.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* productExceptSelf(int* nums, int numsSize, int* returnSize) {
    int left[numsSize];
    int right[numsSize];
    int l_mult = 1;
    int r_mult = 1;

    for(int i=0;i<numsSize;i++){
        int j = numsSize-1-i;
        left[i] = l_mult;
        right[j] = r_mult;
        l_mult *= nums[i];
        r_mult *= nums[j];
    }
    int* res = (int*)malloc((numsSize+1)*sizeof(int));
    for(int i=0;i<numsSize;i++){
        res[i] = left[i] * right[i];
    }
    *returnSize = numsSize;
    return res;
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int total = 1;
        int zeros = 0;

        for(int i = 0 ; i < nums.length; i++){
            if(nums[i] != 0) total *= nums[i];
            else zeros++;
        }

        int[] result = new int[nums.length];
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0 && zeros <= 1) result[i] = total;
            else if(zeros == 0)            result[i] = total/nums[i];
        }
        return result;
    }
}